### ISSUE TITLE
Make `Model#create()` return `Promise<mongoose.Document>` rather than just `Promise`

### ISSUE THREAD

**Issue by Skyclo** [Automattic/mongoose#10283]
**Do you want to request a *feature* or report a *bug*?**
Feature

**What is the current behavior?**
Currently, `Model#create( )` only returns the type `Promise`.

**What is the expected behavior?**
`Model#create( )` should return a promise with a document included so the document can be manipulated after creation. For example, lets say that a developer wants to create a new document for the User schema and the ID for the user is created when the document is created (in the schema definition). 
```js
// server.js
// ...
await Users.create({ // Returns a promise but no document
    firstName: "Tom",
    lastName: "Smith"
})
let userInformation = Users.findOne({lastName:  "Smith"})
userInformation.lastName = "Miller"
userInformation.save()
// ...
```

```js
// Users.js
const mongoose = require('mongoose')
const { v4: uuidv4 } = require('uuid') // Or any other ID-like system

const UserSchema = new mongoose.Schema({
    uid: {
        type: String,
        required: true,
        default: uuidv4() // User ID is generated upon document creation
    },
    firstName: {
        type: String,
        required: true
    },
    lastName: {
        type: String,
        required: true
    }
})

module.exports = mongoose.model('users', UserSchema)
```
In the example above, the developer would have to try and retrieve the information immediately after creating it if they wanted, for example, to change the last name. They would have to query for the document, increasing load on the server and DB. If `Model#create( )` returned the document upon creation, they wouldn't have to do this. See below how this would possibly work.

```js
// server.js
// ...
let userInformation = await Users.create({ // Returns a promise with the document
    firstName: "Tom",
    lastName: "Smith"
})
userInformation.lastName = "Miller"
userInformation.save()
// ...
```

With this implementation, the developer can be sure that they query for the right information. If a specific key is needed to index/query a document (ex: the UID key above), and that key value is unknown to the developer at the time of creation, issues can arise. (Ex: there are multiple "Tom Smith"s).

tl;dr: I am proposing that `Model#create( )` returns `Promise<mongoose.Document>` rather than just `Promise`.

Thanks in advance!

**Comment by AbdelrahmanHafez** [comment]
@Skyclo 
If you need the document immediately, how about using
```js
const user = new User({ // immediately available in memory, not saved to the database
    firstName: "Tom",
    lastName: "Smith"
});

user.lastName = "Miller";
await user.save();
```

Under the hood, `Model.create(...)` creates a new document like the code snippet above, and `save`s it.
Would this achieve what you're trying to achieve?

**Comment by Skyclo** [comment]
> @Skyclo 
> If you need the document immediately, how about using
> ```js
> const user = new User({ // immediately available in memory, not saved to the database
>     firstName: "Tom",
>     lastName: "Smith"
> });
> 
> user.lastName = "Miller";
> await user.save();
> ```
> 
> Under the hood, `Model.create(...)` creates a new document like the code snippet above, and `save`s it.
> Would this achieve what you're trying to achieve?

Hello @AbdelrahmanHafez,

I saw that in the documentation but I was unsure if any other properties were generated (for example, the default value generated in the schema or the `_id` key generated by mongo). If this does that then it works for me! I just thought of proposing the idea to save space and time.

Thanks!

**Comment by AbdelrahmanHafez** [comment]
@Skyclo 
In both `User.create(...)` and `new User(...)` the `_id` is generated on the application side, and then is sent to MongoDB. For this purpose `new User` should achieve exactly what you need.

I'd try to avoid polluting a promise with custom properties, and just let it resolve when saved successfully. Polluting it will potentially create an unneeded high maintenance cost in the future.

**Comment by Skyclo** [comment]
> @Skyclo 
> In both `User.create(...)` and `new User(...)` the `_id` is generated on the application side, and then is sent to MongoDB. For this purpose `new User` should achieve exactly what you need.
> 
> I'd try to avoid polluting a promise with custom properties, and just let it resolve when saved successfully. Polluting it will potentially create an unneeded high maintenance cost in the future.

Ah, I understand. Thanks! Have a nice day.

**Comment by Dodgingrain1** [comment]
I guess I don't understand why:
newuser = await User.create(...) 

doesn't return the User object with everything populated after the database insert, isn't that the point of await?  Having to create the user and then take a second step to save it seems verbose and create doesn't seem to have much value.

**Comment by AbdelrahmanHafez** [comment]
Looking at this a couple of years later I have no idea what we were discussing. Now that I read it, it feels like the author meant TS types, and my comments addressed some runtime issues instead.

@Dodgingrain1  I wrote a quick test in TS and types seem to be working as I would expect them.
<img width="939" alt="Screenshot 2023-06-24 at 2 16 56 AM" src="https://github.com/Automattic/mongoose/assets/19984935/27273cd0-8045-4c51-a101-067415258d81">


We've done a lot of work on mongoose in the past two years, and any information in this issue is probably outdated. If you're still facing any issues, please open a new issue following the issue template and we'd be more than happy to help.
